packages/cozy-konnector-libs/src/libs/hydrateAndFilter.js

```diff
@@ -107,7 +107,18 @@ const hydrateAndFilter = (documents = [], doctype, options = {}) => {
 
     log('debug', selector, 'selector')
 
-    return cozy.data.query(index, { selector })
+    const result = []
+    while (true) {
+      log('debug', `Fetching next db items ${result.length}`)
+      const resp = await cozy.data.query(index, {
+        selector,
+        wholeResponse: true,
+        skip: result.length
+      })
+      result.push.apply(result, resp.docs)
```

Comment:
it might be easier to read using [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax):
```js
result.push(...resp.docs);
```